pkg/api/api.go

```diff
@@ -381,6 +381,9 @@ func (hs *HTTPServer) registerRoutes() {
 		// DataSource w/ expressions
 		apiRoute.Post("/ds/query", authorize(reqSignedIn, ac.EvalPermission(ActionDatasourcesQuery)), routing.Wrap(hs.QueryMetricsV2))
 
+		// Validated query
+		apiRoute.Post("/dashboards/id/:dashboardId/panels/:panelId/query", authorize(reqSignedIn, ac.EvalPermission(ActionDatasourcesQuery)), routing.Wrap(hs.QueryMetricsFromDashboard))
```

Comment:
I think we generally prefer going with UIDs over IDs, it looks like GetDashboard can work by uid, do we need to be using id?

Reply:
We didn't know that. Happy to make the change!